roles/matrix-bridge-mautrix-discord/tasks/setup_install.yml

```diff
@@ -0,0 +1,149 @@
+    source: "{{ 'pull' if ansible_version.major > 2 or ansible_version.minor > 7 else omit }}"
+    force_source: "{{ matrix_mautrix_discord_docker_image_force_pull if ansible_version.major > 2 or ansible_version.minor >= 8 else omit }}"
+    force: "{{ omit if ansible_version.major > 2 or ansible_version.minor >= 8 else matrix_mautrix_discord_docker_image_force_pull }}"
+  when: not matrix_mautrix_discord_container_image_self_build
+  register: result
+  retries: "{{ matrix_container_retries_count }}"
+  delay: "{{ matrix_container_retries_delay }}"
+  until: result is not failed
+
+- name: Ensure Mautrix discord repository is present on self-build
+  ansible.builtin.git:
+    repo: "{{ matrix_mautrix_discord_container_image_self_build_repo }}"
+    dest: "{{ matrix_mautrix_discord_docker_src_files_path }}"
+    version: "{{ matrix_mautrix_discord_container_image_self_build_branch }}"
+    force: "yes"
+  become: true
+  become_user: "{{ matrix_user_username }}"
+  register: matrix_mautrix_discord_git_pull_results
+  when: "matrix_mautrix_discord_container_image_self_build | bool"
+
+- name: Ensure Mautrix discord Docker image is built
+  docker_image:
+    name: "{{ matrix_mautrix_discord_docker_image }}"
+    source: build
+    force_source: "{{ matrix_mautrix_discord_git_pull_results.changed if ansible_version.major > 2 or ansible_version.minor >= 8 else omit }}"
+    force: "{{ omit if ansible_version.major > 2 or ansible_version.minor >= 8 else matrix_mautrix_discord_git_pull_results.changed }}"
+    build:
+      dockerfile: Dockerfile
+      path: "{{ matrix_mautrix_discord_docker_src_files_path }}"
+      pull: true
+  when: "matrix_mautrix_discord_container_image_self_build | bool"
+
+- name: Check if an old database file exists
+  ansible.builtin.stat:
+    path: "{{ matrix_mautrix_discord_base_path }}/mautrix-discord.db"
+  register: matrix_mautrix_discord_stat_database
+
+- name: Check if an old matrix state file exists
+  ansible.builtin.stat:
+    path: "{{ matrix_mautrix_discord_base_path }}/mx-state.json"
+  register: matrix_mautrix_discord_stat_mx_state
+
+- name: (Data relocation) Ensure matrix-mautrix-discord.service is stopped
+  ansible.builtin.service:
+    name: matrix-mautrix-discord
+    state: stopped
+    enabled: false
+    daemon_reload: true
+  failed_when: false
+  when: "matrix_mautrix_discord_stat_database.stat.exists"
+
+- name: (Data relocation) Move mautrix-discord database file to ./data directory
+  ansible.builtin.command: "mv {{ matrix_mautrix_discord_base_path }}/mautrix-discord.db {{ matrix_mautrix_discord_data_path }}/mautrix-discord.db"
+  when: "matrix_mautrix_discord_stat_database.stat.exists"
+
+- name: (Data relocation) Move mautrix-discord mx-state file to ./data directory
+  ansible.builtin.command: "mv {{ matrix_mautrix_discord_base_path }}/mx-state.json {{ matrix_mautrix_discord_data_path }}/mx-state.json"
+  when: "matrix_mautrix_discord_stat_mx_state.stat.exists"
+
```

Comment:
```suggestion
```

These are old data migration tasks (from when the mautrix whatsapp `.db` file was stored in the base path, not in the data path). It's not needed here.